middleware/reverse/setup.go

```diff
@@ -117,8 +121,12 @@ func reverseParse(c *caddy.Controller) (nets networks, fall bool, err error) {
 				if ipnet.IP.To4() == nil {
 					regexIP = regexMatchV6
 				}
+				prefix := "^"
+				if wildcard {
+					prefix += ".*"
```

Comment:
nit - Removing the explicit ^ should work without needing to specify the .* at the beginning, so we can just set prefix to the empty string.